### DISCUSSION TITLE
Blobfuse2 with rhel8 start/stop using systemd service we notice permission issues.

### DISCUSSION THREAD

**Discussion by naisu07-us** [Azure/azure-storage-fuse#1565] [Q&A] [answered]
Manually mounting the blob file system on RHEL 8 works with blobfuse2, but when using a systemd service with a specific user, we encounter with no permission with mounted file systems and log files give error as 
[blobfuse2-systemd-service-error.txt](https://github.com/user-attachments/files/17736303/blobfuse2-systemd-service-error.txt)
a libfuse destruction error.

Attaching file with all config, systemd service file and error logs.   
Please help with this issues.

Thanks,
-Sunil.

**Comment by naisu07-us** [comment]
Blobfuse2 mounts the file system but it give permission error on that file system.

$ ls -l
**ls: cannot access 'myfuse': Permission denied**
total 8
drwxr-xr-x  2 myuserid mygrp  33 Nov 13 17:38 azure-storage-fuse
-rwxr-xr-x  1 myuserid mygrp 755 Nov 13 02:47 blobfuse-startstop.sh
-rw-r--r--. 1 myuserid mygrp 576 Nov 12 23:46 fuse_connection_v1.yml
**d?????????? ? ?       ?        ?            ? myfuse**
drwxr-xr-x. 4 myuserid mygrp  34 Oct 30 21:58 rpmbuild
drwxr-xr--  2 myuserid mygrp  38 Nov 13 02:13 tmp
drwxr-xr-x  2 myuserid mygrp   6 Nov 13 17:11 tmpblobfuse

**Comment by bhansalivikas** [comment]
If you are using systemd or /etc/fstab mode where mount is done automatically you need to use "-o allow_other" option in your mount as the mount is not done by the user trying to access the path. In such case other users are not allowed to access it. This option will allow you to access the mount done by any other user in the system.

**Comment by sanuskaria123** [comment]
@vibhansa-msft : Thank you for the response.  Using systemd to mount with "-o allow_other" option gives full permission (0777) to other user and groups. Is there any way we can limit other users with just read permission while mounting using systemd?

**Comment by bhansalivikas** [comment, marked as answer]
Only other way is to somehow mount via the user who wants to assume the ownership of the mount path.

**Reply by naisu07-us** [reply]
This issue is now fixed, we added user:group permission to blobfuse2 mount and that fixed the issue, but while mounting blobfuse2 mount folder it adds default permission to mount directory as 077.  
Thanks,

**Reply by bhansalivikas** [reply]
Mount directory permissions can not be altered anything inside the mount path is what blobfuse controls. As you have provided allow_other default behaviour is to set permissions to 777 for all files and folders. If its HNS account this can be controlled by setting the permissions on the files and folders

**Reply by tspearconquest** [reply]
You can fake the permissions. If you set `-o uid=` and `-o gid=` and `-o umask=` with the correct values. `uid=` should be the uid of the user to mount the storage as, `gid=` should be the gid of the user to mount the storage as, and `umask=` should be 177 to make the permissions appear as 600 for files and directories. This does not enforce the permissions; if you leave `-o allow_other` after adding this, other users can still write to files and folders, but it at least no longer appears as 777 permissions to any tools which will scan the filesystem permissions and report a risk that isn't valid.
